src/basic/terminal-util.c

```diff
@@ -455,7 +455,7 @@ int acquire_terminal(
                                         goto fail;
                                 }
 
-                                r = fd_wait_for_event(fd, POLLIN, ts + timeout - n);
+                                r = fd_wait_for_event(notify, POLLIN, ts + timeout - n);
```

Comment:
So, i am not sure i grok the endless loop you are seeing. Note that before trying to open the device, we explicitly flush out all queued inotify events with flush_fd(). Thus, our own close() from an earlier event doesn't matter, we won't see it on the inotify queue anymore...

Can you provide an "strace" of this endless loop you are seeing?